main/src/com/google/refine/commands/EngineDependentCommand.java

```diff
@@ -58,29 +52,15 @@ SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
  * Note that there are interactions on the client side that change only individual cells or individual rows (such as
  * starring one row or editing the text of one cell). These interactions do not depend on the faceted browsing engine's
  * configuration, and so they don't invoke commands that subclass this class. See AnnotateOneRowCommand and
- * EditOneCellCommand as examples.
+ * EditOneCellCommand as examples. Conversely, there are commands like GetRowsCommand and ExportRowsCommand which use
+ * the engine config, but don't modify the project, which don't subclass this.
  */
-abstract public class EngineDependentCommand extends Command {
-
-    @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response)
-            throws ServletException, IOException {
-        if (!hasValidCSRFToken(request)) {
-            respondCSRFError(response);
-            return;
-        }
-
-        try {
-            Project project = getProject(request);
-
-            AbstractOperation op = createOperation(project, request, getEngineConfig(request));
-            Process process = op.createProcess(project, new Properties());
+abstract public class EngineDependentCommand extends OperationCommand {
```

Comment:
Changing the class hierarchy here likely breaks binary compatibility, so needs to be rethought.